### PR TITLE
[Test Fix] LPD-23320 Fix NewEnvJVMTestRuleTest deserializing blank value as null

### PR DESCRIPTION
**Pull request by julschong**
This is for issue: https://liferay.atlassian.net/browse/LPD-23320
Related Testray failure: https://testray.liferay.com/home/-/testray/case_results/2556478810

Currently, null and blank would both be deserialized to be null. This is causing this test to fail comparison when environment contains blank property because it becomes null on the other end.

This PR serialize null as "null" string so that we can differentiate between null and blank "" when we deserialize

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

File: portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java [hunk 1]
```diff
@@ -228,7 +229,9 @@ private static String _toString(Map<String, String> map) {
 		for (Map.Entry<String, String> entry : map.entrySet()) {
 			sb.append(entry.getKey());
 			sb.append(_SEPARATOR_KEY_VALUE);
-			sb.append(entry.getValue());
+			sb.append(
+				(entry.getValue() == null) ? _NULL_PLACE_HOLDER :
```
Comment (julschong) [inline portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java:233]: this is changed to using _NULL_PLACE_HOLDER

File: portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java [hunk 2]
```diff
@@ -244,10 +247,12 @@ private Map<String, String> _fromString(String s) {
...
 
 			if (parts.length == 1) {
-				map.put(parts[0], null);
+				map.put(parts[0], StringPool.BLANK);
 			}
 			else {
-				map.put(parts[0], parts[1]);
+				map.put(
+					parts[0],
+					parts[1].equals(_NULL_PLACE_HOLDER) ? null : parts[1]);
```
Comment (julschong) [inline portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java:255]: this is changed to using _NULL_PLACE_HOLDER

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
added some more detail in commit message and change null holder to "_NULL_HOLDER_" following similar convention as other static variables

**Comment by julschong** [comment]
ci:report:2560992938

**Comment by julschong** [comment]
ci:report:2561010456

**Comment by julschong** [comment]
unique failure `com.liferay.document.library.internal.search.spi.model.index.contributor.test.DLFileEntryModelDocumentContributorTest` should be unrelated

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/149631

**Pull request closed without merging**